x/superfluid/keeper/stake.go

```diff
@@ -503,17 +504,31 @@ func (k Keeper) forceUndelegateAndBurnOsmoTokens(ctx sdk.Context,
 	if err != nil {
 		return err
 	}
-	// TODO: Better understand and decide between ValidateUnbondAmount and SharesFromTokens
-	// briefly looked into it, did not understand what's correct.
-	// TODO: ensure that intermediate account has at least osmoAmount staked.
+
 	shares, err := k.sk.ValidateUnbondAmount(
 		ctx, intermediaryAcc.GetAccAddress(), valAddr, osmoAmount,
 	)
-	if err == stakingtypes.ErrNoDelegation {
-		return nil
```

Comment:
We still want this check right?

Reply:
yes, added it back in the latest commit. Thanks for the catch 🙏